artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/QueueInfo.java

```diff
@@ -107,7 +108,11 @@ public void incrementConsumers() {
    }
 
    public void decrementConsumers() {
-      numberOfConsumers--;
+      if (numberOfConsumers > 0) {
```

Comment:
This isnt being atomically updated so still possible two decrements at the same time can cause the number to go negative incorrectly.

To avoid this ideally should syncronize any update to the field or make changes to the field using an atomic updater

Reply:
@michaelandrepearce +1.  I will make this an AtomicInteger cheers.

Reply:
The atomic updater would be better then atomic integer.  I think that’s what Michael is referring.